File: frontend/lib/model/judgement/userankrelativejudgement/command/EditUseRankRelativeJudgementCommand.ts
import IEditJudgementCommand from "../../command/IEditJudgementCommand";

export default class EditUseRankRelativeJudgementCommand implements IEditJudgementCommand {

    public readonly owner: string;
    public readonly project: string;
    public readonly phase: string;

    public readonly instance: string;
    public readonly annotator: string;

    public readonly UUID: string;

    public readonly label: string;
    public readonly comment: string;

    constructor(owner: string, project: string, phase: string, instance: string, annotator: string, UUID: string, label: string, comment: string) {
        this.owner = owner;
        this.project = project;
        this.phase = phase;
        this.instance = instance;
        this.annotator = annotator;
        this.UUID = UUID;
        this.label = label;
        this.comment = comment;
    }


}

File: backend/src/main/java/de/garrafao/phitag/application/judgement/userankrelativejudgement/data/EditUseRankRelativeJudgementCommand.java
package de.garrafao.phitag.application.judgement.userankrelativejudgement.data;

import de.garrafao.phitag.application.judgement.data.IEditJudgementCommand;
import lombok.Getter;

@Getter
public class EditUseRankRelativeJudgementCommand implements IEditJudgementCommand {

    private String owner;
    private String project;
    private String phase;

    private String instance;
    private String annotator;
    
    private String UUID;

    private String label;
    private String comment;

    public EditUseRankRelativeJudgementCommand(String owner, String project, String phase, String instance, String annotator, String UUID, String label, String comment) {
        this.owner = owner;
        this.project = project;
        this.phase = phase;

        this.instance = instance;
        this.annotator = annotator;
        
        this.UUID = UUID;
        
        this.label = label;
        this.comment = comment;
    }
    
}
